File: year_3/semester_6/PS/tsma/spa/src/app/model/dtos/calendar/calendar-dto.ts
import {ShiftSequence} from "./calendar-create-dto";

export class CalendarDTO {

	readonly id  : number
	title       : string
	description : string

	startDate   : Date
	endDate     : Date

	sequences : ShiftSequence[]

	constructor(id:number,
				title:string,
				description:string,
				startDate:Date,
				endDate:Date,
				sequences:ShiftSequence[]) {
		this.id = id
		this.title = title
		this.description = description
		this.startDate = startDate
		this.endDate = endDate
		this.sequences = sequences
	}
}

File: year_3/semester_6/PS/tsma/api/src/main/java/pt/isel/tsma/entity/dto/calendar/CalendarDTO.java
package pt.isel.tsma.entity.dto.calendar;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import pt.isel.tsma.entity.dto.shift.ShiftDurationDTO;
import pt.isel.tsma.entity.model.shift.Type;

import java.time.LocalDate;
import java.util.HashMap;
import java.util.List;

@Data
@NoArgsConstructor
public class CalendarDTO implements ICalendarDTO {

	private long id;

	@NonNull
	private String title;

	private String description;

	@NonNull
	private LocalDate startDate;

	@NonNull
	private LocalDate endDate;

	@NonNull
	private HashMap<Type, ShiftDurationDTO> shiftDurations;

	@NonNull
	private List<SequenceDTO> sequences;
}
